### DISCUSSION TITLE
Stock item status as attribute of location

### DISCUSSION THREAD

**Discussion by turist79** [inventree/InvenTree#8415] [Ideas]
If stock location could have stock item status assigned to it (Damaged, Quarantined etc) it would improve the workflow allowing to create specialized locations (bins) like "holding" bin (implemented in Openboxes app), or "repair location" (for damaged items), or "scrapping" bin, where any item transferred to such location would get the corresponding status automagically.

**Comment by SchrodingersGat** [comment]
@turist79 nice idea. This could be easily implemented via a plugin and I wonder if this might be the way to go - this would allow flexibility in the "action" to occur on a stock transfer.

- We already have "location types" which could be used to identify a particular location to perform an action on
- Actions such as "transfer_stock_item" are already supported and can be intercepted by plugins

**Reply by turist79** [reply]
> @turist79 nice idea. This could be easily implemented via a plugin and I wonder if this might be the way to go - this would allow flexibility in the "action" to occur on a stock transfer.
> 
> * We already have "location types" which could be used to identify a particular location to perform an action on
> * Actions such as "transfer_stock_item" are already supported and can be intercepted by plugins

this sounds like not a very complicated type of plugin to start with. Should give it a try probably :)

**Reply by SchrodingersGat** [reply]
@turist79 LMK if you need any pointers or run into any issues! if you get something working, we can add it to our plugins list - https://inventree.org/plugins.html
